sharder/deterministic.go

```diff
@@ -122,28 +123,50 @@ func (d *DeterministicSharder) Start() error {
 		if err != nil {
 			return errors.Wrap(err, "failed to get local interface list to initialize sharder")
 		}
+		localAddrCidrs := make([]string, len(localAddrs))
+		for i, addr := range localAddrs {
+			localAddrCidrs[i] = addr.String()
+		}
+
+		// If RedisIdentifier is an IP, add it to localAddrs.
+		redisIdentifier, err := d.Config.GetRedisIdentifier()
+		if err == nil {
+			d.Logger.Debug().Logf("Using RedisIdentifier as self: %s", redisIdentifier)
+			if ip := net.ParseIP(redisIdentifier); ip != nil {
+				d.Logger.Debug().Logf("RedisIdentifier is an IP")
+				localAddrCidrs = append(localAddrCidrs, fmt.Sprintf("%s/32", redisIdentifier))
```

Comment:
Assumes that RedisIdentifier is IPv4. That's maybe not a valid assumption? Should this Sprintf be conditional? And should it be smarter than just using `/32`?

Reply:
The network portion isn't used in the following logic.  It's just extraneous data from `net.InterfaceAddrs()` that has to be replicated by this logic.  Maybe the solution is to strip the network portion earlier?  I'll give it a shot.

Reply:
There, take a look now.